db/view/view.cc

```diff
@@ -1794,9 +1834,9 @@ get_view_natural_endpoint(
     // right now but to give up attempt to send the update or write a hint
     // to the paired, permanently down replica.
     if (!view_topology.get_node(replica).left()) {
-        return replica;
+        return std::make_pair(std::move(replica), std::move(extra_view_endpoint));
     } else {
-        return std::nullopt;
+        return std::make_pair(std::nullopt, std::move(extra_view_endpoint));
     }
```

Comment:
shouldn't we perform the same check for deciding whether to push a update to extra view replica, that is under the condition being handled here?

Reply:
Right, we probably should. The endpoint which got the new replica of the view might just be the one we're replacing. I added the same workaround as for the paired endpoint in the rebase